File: tools/seatunnel-python-sdk/seatunnel/endpoints/jobs.py
#  Licensed to the Apache Software Foundation (ASF) under one or more
#  contributor license agreements.  See the NOTICE file distributed with
#  this work for additional information regarding copyright ownership.
#  The ASF licenses this file to You under the Apache License, Version 2.0
#  (the "License"); you may not use this file except in compliance with
#  the License.  You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.


import json
from typing import Optional

from ..client import Client
from ..helpers.httpMethod import HttpMethod
from ..helpers.jobStatus import JobStatus
from ..helpers.queryParams import (
    StopJobQueryParams,
    SubmitJobFileQueryParams,
    SubmitJobQueryParams,
)


class JobsApi:
    def __init__(self, client: Client):
        self.client = client

    def submit_job(self, conf: str, params: SubmitJobQueryParams):
        """
        Submit A Job
        """
        return self.client.request(
            HttpMethod.POST, "/submit-job", content=conf, params=params.__dict__
        )

    def submit_job_from_file(self, filePath: str, params: SubmitJobFileQueryParams):
        """
        Submit A Job By Upload Config File
        """
        with open(filePath, "rb") as file:
            return self.client.request(
                HttpMethod.POST,
                "/submit-job/upload",
                files={"config_file": file},
                params=params.__dict__,
            )

    def submit_jobs(self, confs: str):
        """
        Batch Submit Jobs
        """
        return self.client.request(HttpMethod.POST, "/submit-jobs", content=confs)

    def stop_job(self, params: StopJobQueryParams):
        """
        Stop A Job
        """
        return self.client.request(HttpMethod.POST, "/stop-job", content=json.dumps(params.__dict__))

    def stop_jobs(self, params: list[StopJobQueryParams]):
        """
        Batch Stop Jobs
        """
        return self.client.request(
            HttpMethod.POST, "/stop-jobs", content=json.dumps([obj.__dict__ for obj in params])
        )

    def get_running_jobs(self):
        """
        Returns An Overview And State Of All Jobs
        """
        return self.client.request(HttpMethod.GET, "/running-jobs")

    def get_job_details(self, jobId: int):
        """
        Return Details Of A Job
        """
        return self.client.request(HttpMethod.GET, f"/job-info/{jobId}")

    def get_finished_jobs_info(self, state: Optional[JobStatus] = None):
        """
        Return Details Of A Job
        """
        jobStatus = "" if state is None else state.value
        return self.client.request(HttpMethod.GET, f"/finished-jobs/{jobStatus}")

File: seatunnel-engine/seatunnel-engine-ui/src/locales/en_US/jobs.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export default {
  runningJobs: 'Running Jobs',
  finishedJobs: 'Finished Jobs',
  operations: {
    title: 'Submit Job',
    textSubmit: 'Config Text',
    fileSubmit: 'Config File',
    jobName: 'Job Name',
    jobNamePlaceholder: 'Optional job name',
    configFormat: 'Format',
    configContent: 'Config',
    configPlaceholder: 'Paste JSON, HOCON, or SQL job config',
    configFile: 'Config File',
    restore: 'Restore',
    restoreMode: 'Restore Mode',
    restoreSourceJobId: 'Restore Source Job ID',
    restoreSourceJobIdPlaceholder: 'Existing job ID containing the selected state',
    chooseFile: 'Choose File',
    reset: 'Reset',
    submit: 'Submit',
    configRequired: 'Job config is required.',
    fileRequired: 'Please choose a config file.',
    restoreSourceJobIdRequired: 'Restore source job ID is required when restore is enabled.',
    submitSuccess: 'Submitted {job}.',
    submitFailed: 'Failed to submit job.',
    submitOutcomeUnknown: 'The submit result is unknown. Refresh jobs before submitting again.',
    confirm: 'Confirm',
    cancelConfirm: 'Dismiss',
    submitConfirmMessage: 'Submit this job?'
  },
  actions: {
    view: 'View',
    stop: 'Stop',
    savepoint: 'Savepoint',
    cancel: 'Cancel',
    confirm: 'Confirm',
    cancelConfirm: 'Dismiss',
    stopConfirmMessage: 'Stop {job} gracefully?',
    savepointConfirmMessage: 'Stop {job} with savepoint?',
    cancelConfirmMessage: 'Force cancel {job}?',
    stopSuccess: 'Stop request sent for {job}.',
    stopFailed: 'Failed to stop {job}.',
    savepointSuccess: 'Savepoint stop request sent for {job}.',
    savepointFailed: 'Failed to stop {job} with savepoint.',
    cancelSuccess: 'Cancel request sent for {job}.',
    cancelFailed: 'Failed to cancel {job}.',
    operationOutcomeUnknown: 'The operation result for {job} is unknown. Refresh before retrying.',
    refreshFailed: 'Failed to refresh running jobs.'
  }
}
